server/controllers/auth/authentication.js

```diff
@@ -0,0 +1,10 @@
+const { jwtVerify } = require('../../helpers/jwtVerifyPromise');
+
+const key = process.env.KEY;
+exports.auth = (req, res, next) => {
+  const { token } = req.cookies;
```

Comment:
😮 What if there is no token in the request header? This will throw an error since you can't destructure sth that does not exist!
So, you should check if there are a cookie header and a token property in the header object. If yes, then verify the token
If there is no token, you should throw a 401 error.
```javascript
if(req.cookies && req.cookies.token){
jwtVerify(...)
} else{
next({code: 401})
}
```
I think the status code here is 401 not 403. 401 is for authorization, like the user is not logged in or his cookie is manipulated or his password is not correct. But 403 means forbidden, which means the user might be logged in but is trying to request a route, available in your api but he does not have the permission to access it like endpoints for admin.

Do more research on this topic and let me know what you find!